lib/components/base/CopyCode.vue

```diff
@@ -7,28 +7,21 @@
 </template>
 
 <script setup>
+import { ref } from 'vue'
 import { CheckIcon, ClipboardCopyIcon } from '@'
-</script>
 
-<script>
-export default {
-  props: {
-    text: {
-      type: String,
-      required: true,
-    },
-  },
-  data() {
-    return {
-      copied: false,
-    }
-  },
-  methods: {
-    async copyText() {
-      await navigator.clipboard.writeText(this.text)
-      this.copied = true
-    },
+const props = defineProps({
+  text: {
+    type: String,
+    required: true,
   },
+})
```

Comment:
I would honestly suggest switching to TS and simply define props via a generic:

```suggestion
const props = defineProps<{ text: string }>()
```